plugins/embed-optimizer/detect.js

```diff
@@ -0,0 +1,90 @@
+const consoleLogPrefix = '[Embed Optimizer]';
```

Comment:
Could you add a doc block on top of this file to explain what its purpose is? I have a hard time deciphering how it fits into the functionality as there is only very minimal documentation.

I was expecting for JS related functionality to be already covered by Optimization Detective's own detection script, which is why I'm wondering about this.